### ISSUE TITLE
Can't remove special vdev (wrong block size, system reboot)

### ISSUE THREAD

**Issue by bullsei** [openzfs/zfs#10952]
Distribution Name	| TrueNAS
Distribution Version	| TrueNAS-12.0-RC1
Linux Kernel	|
Architecture	| BSD
ZFS Version: zfs-2.0.0-rc1
zfs-kmod-v2020082800-ZoF_735d1ffe9

SPL Version: 	| both commands don't work

### Describe the problem you're observing

I added a special metadata vdev consisting of 2 mirrored SSDs to my pool, afterwards i found out its not really what i wanted and i want my SSDs in a seperate pool, so i tried to remove them via zpool remove, which should be possible as written here: https://openzfs.github.io/openzfs-docs/man/8/zpool-remove.8.html

after i executed that command the system rebooted (possible crashed?)
after the reboot the vdev is still there. i tried to remove it again but this time i get the following error: 
**cannot remove mirror-3: invalid config; all top-level vdevs must have the same sector size and not be raidz.**

zpool status shows this currently:

```
 pool: mypool
 state: ONLINE
status: One or more devices are configured to use a non-native block size.
	Expect reduced performance.
action: Replace affected devices with devices that support the
	configured block size, or migrate data to a properly configured
	pool.
  scan: scrub repaired 0B in 18:42:54 with 0 errors on Mon Sep 14 20:43:23 2020
remove: Removal of vdev 0 copied 1.78T in 5h53m, completed on Sat Feb 22 23:10:53 2020
    45.8M memory used for removed device mappings
config:

	NAME                                            STATE     READ WRITE CKSUM
	mypool                                          ONLINE       0     0     0
	  mirror-1                                      ONLINE       0     0     0
	    gptid/8b511c10-19df-11ea-986b-ac1f6bf61788  ONLINE       0     0     0
	    gptid/78a51654-19e3-11ea-986b-ac1f6bf61788  ONLINE       0     0     0
	  mirror-2                                      ONLINE       0     0     0
	    gptid/1bceea58-75a8-11ea-b748-001b21bbeda0  ONLINE       0     0     0
	    da0                                         ONLINE       0     0     0
	special	
	  mirror-3                                      ONLINE       0     0     0
	    gptid/7aebf682-fa7b-11ea-acf8-001b21bbeda0  ONLINE       0     0     0
	    gptid/7b40d98a-fa7b-11ea-acf8-001b21bbeda0  ONLINE       0     0     0
```

Is there anything i could do to fix that problem?

I am not sure how that could be reproduced exactly on another system, maybe the same way?

### Include any warning/errors/backtraces from the system logs
Not sure what kind of logs would be useful here, please let me know

**Comment by gmelikov** [comment]
Not a bug, from same link https://openzfs.github.io/openzfs-docs/man/8/zpool-remove.8.html

> When the primary pool storage includes a top-level raidz vdev only hot spare, cache, and log devices can be removed.

Unfortunately, for now `zpool remove` doesn't work on pools with raidz vdevs.

**Comment by bullsei** [comment]
My pool doesnt have a raidz vdev. My pool is made of 2 mirrored vdevs. So in my opinion it should work?

**Comment by gmelikov** [comment]
@bullsei Ah, my bad! You're right, reopen.

**Comment by bullsei** [comment]
Thanks, if i can provide any logs or data please let me know. I am not sure what would be helpful for you and how i would collect that data.

**Comment by behlendorf** [comment]
@bullsei this sounds like it may be a duplicate of #10932.  The fix was included in the zfs-2.0.0-rc2 tag

**Comment by h1z1** [comment]
Was this ever confirmed fixed?  10932 looks different.  I see @gmelikov reopened it on Sept 19, 2020 but github shows it closed??  I don't see where that happened.

Regardless somehow I've been able to trigger it.  Similar problem with a 3-way mirror though not quite exact..  

For whatever reason deadman is tripping on it which never recovers.  It trips causing z_rd_int to deadlock on something.  The entire host ends up fsck itself... It's a special vdev (no comment), zpool throws the same error though

```
# zpool remove pinky /home/pinky.special.00
cannot remove /home/pinky.special.00: invalid config; all top-level vdevs must have the same sector size and not be raidz.
# 
```

It isn't in a raidz.

```
  pool: pinky
 state: ONLINE
status: One or more devices has experienced an unrecoverable error.  An
        attempt was made to correct the error.  Applications are unaffected.
action: Determine if the device needs to be replaced, and clear the errors
        using 'zpool clear' or replace the device with 'zpool replace'.
   see: https://openzfs.github.io/openzfs-docs/msg/ZFS-8000-9P
  scan: scrub in progress since Thu May  5 17:48:11 2022
        760G scanned at 286M/s, 432G issued at 162M/s, 8.17T total
        136K repaired, 5.16% done, 13:53:59 to go
remove: Removal of vdev 6 copied 1.68G in 0h0m, completed on Tue May  3 05:43:49 2022
        4.45K memory used for removed device mappings
config:

        NAME                                   STATE     READ WRITE CKSUM
        pinky                                   ONLINE       0     0     0
          mirror-0                             ONLINE       0     0     0
            ata-HGST_HDN721010ALE604  ONLINE       0     0     5  (repairing)
            ata-HGST_HDN721010ALE604  ONLINE       0     0     2  (repairing)
            ata-HGST_HDN721010ALE604  ONLINE       0     0     6  (repairing)
        special 
          /home/pinky.special.00                 ONLINE       0     0     0
        logs
          nvme1n0-part7      ONLINE       0     0     0
        cache
          nvme0n1p8                            ONLINE       0     0     0
          nvme0n2p8                            ONLINE       0     0     0

errors: No known data errors
# zfs -V
zfs-2.1.99-1
zfs-kmod-2.1.99-1
#
```

Clearly the pool is not in a happy place.
